common/src/main/java/com/wynntils/core/net/athena/WynntilsAccountManager.java

```diff
@@ -86,55 +82,42 @@ public void dumpEncodedConfig(String name) {
-                .postJsonElement(authParams)
-                .handleJsonObject(json -> {
-                    if (!json.has("authToken")) return false;
-                    token = json.get("authToken").getAsString(); /* md5 hashes*/
-                    JsonObject hashes = json.getAsJsonObject("hashes");
-                    hashes.entrySet()
-                            .forEach((k) -> md5Verifications.put(
-                                    k.getKey(), k.getValue().getAsString())); /* configurations*/
-                    JsonObject configFiles = json.getAsJsonObject("configFiles");
-                    configFiles
-                            .entrySet()
-                            .forEach((k) ->
-                                    encodedConfigs.put(k.getKey(), k.getValue().getAsString()));
-                    loggedIn = true;
-                    WynntilsMod.info("Successfully connected to Athena!");
-                    return true;
-                })
-                .build();
-
-        handler.addAndDispatch(responseEncryption);
+        ApiResponse apiResponse = NetManager.callApi(UrlId.API_ATHENA_AUTH_PUBLIC_KEY);
+        apiResponse.handleJsonObject(json -> {
+            String secretKey = parseAndJoinPublicKey(json.get("publicKeyIn").getAsString());
+
+            Map<String, String> arguments = new HashMap<>();
+            arguments.put("key", secretKey);
+            arguments.put("username", McUtils.mc().getUser().getName());
+            arguments.put("version", String.format("A%s %s", WynntilsMod.getVersion(), WynntilsMod.getModLoader()));
+
+            ApiResponse apiResponse2 = NetManager.callApi(UrlId.API_ATHENA_AUTH_RESPONSE, arguments);
+            apiResponse2.handleJsonObject(json2 -> {
+                token = json2.get("authToken").getAsString(); /* md5 hashes*/
+                JsonObject hashes = json2.getAsJsonObject("hashes");
+                hashes.entrySet()
+                        .forEach((k) ->
+                                md5Verifications.put(k.getKey(), k.getValue().getAsString())); /* configurations*/
+                JsonObject configFiles = json2.getAsJsonObject("configFiles");
+                configFiles
+                        .entrySet()
+                        .forEach((k) ->
+                                encodedConfigs.put(k.getKey(), k.getValue().getAsString()));
+                loggedIn = true;
+                future.complete(true);
+                WynntilsMod.info("Successfully connected to Athena!");
+            });
+        });
+        try {
+            // FIXME: Wait up to 5 seconds for login to complete
+            // This is not ideal, but the rest of the code assumes that login is done
+            // when further down the manager init chain.
+            future.get(5000, TimeUnit.MILLISECONDS);
+        } catch (InterruptedException | ExecutionException | TimeoutException e) {
+            // if timeout is reached, return false
```

Comment:
Log failure?

Reply:
If `loggedIn = true;` is not done at the time we return from this function, we will log an error so it's handled higher above.

I think this mechanism might need some work, but I opted to make minimal changes as part of this patch.